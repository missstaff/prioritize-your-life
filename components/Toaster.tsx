import React from 'react';
import Toast from 'react-native-toast-message';
import AppThemedTouchableOpacity from './app_components/AppThemedTouchableOpacity';

export interface ToasterProps {
  type: 'success' | 'error' | 'info' | 'warning'; // Specific types for better validation
  text1: string;
  text2?: string;
  title: string;
  onPress?: () => void;
}

/**
 * Toaster component.
 * @param {ToasterProps} props - The props for the Toaster component.
 * @returns {JSX.Element} The rendered Toaster component.
 * @component
 * @example
 */
const Toaster: React.FC<ToasterProps> = ({
  onPress,
  type,
  text1,
  text2,
  title,
}) => {
  const showToast = () => {
    Toast.show({
      type,
      text1,
      text2,
    });
  };

  const handleOnPress = () => {
    showToast();
    if (onPress) {
      onPress();
    }
  };

  return (
    <AppThemedTouchableOpacity onPress={handleOnPress} accessibilityLabel={title}>
      {title}
    </AppThemedTouchableOpacity>
  );
};

export default Toaster;
