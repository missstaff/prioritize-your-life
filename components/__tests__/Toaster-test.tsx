import * as React from "react";
import renderer from "react-test-renderer";
import Toaster from "../Toaster";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Toaster type="success" text1="Hello" text2="World" title="Click me" />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("displays error toaster", () => {
  const tree = renderer
    .create(
      <Toaster type="error" text1="Hello" text2="World" title="Click me" />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("displays info toaster", () => {
  const tree = renderer
    .create(
      <Toaster type="info" text1="Hello" text2="World" title="Click me" />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("displays warning toaster", () => {
  const tree = renderer
    .create(
      <Toaster type="warning" text1="Hello" text2="World" title="Click me" />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("displays success toaster without text2", () => {
  const tree = renderer
    .create(<Toaster type="success" text1="Hello" title="Click me" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("displays toaster without text2", () => {
  const tree = renderer
    .create(<Toaster type="success" text1="Hello" title="Click me" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
