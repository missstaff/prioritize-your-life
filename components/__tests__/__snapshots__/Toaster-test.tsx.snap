// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays error toaster 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#CD03EF",
      "borderRadius": 10.714285714285714,
      "marginBottom": 21.428571428571427,
      "opacity": 1,
      "padding": 21.428571428571427,
      "width": "80%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#ECEDEE",
        },
        {
          "fontSize": 34.285714285714285,
          "lineHeight": 51.42857142857143,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        [
          {
            "color": "#fff",
          },
          undefined,
        ],
      ]
    }
  >
    Click me
  </Text>
</View>
`;

exports[`displays info toaster 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#CD03EF",
      "borderRadius": 10.714285714285714,
      "marginBottom": 21.428571428571427,
      "opacity": 1,
      "padding": 21.428571428571427,
      "width": "80%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#ECEDEE",
        },
        {
          "fontSize": 34.285714285714285,
          "lineHeight": 51.42857142857143,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        [
          {
            "color": "#fff",
          },
          undefined,
        ],
      ]
    }
  >
    Click me
  </Text>
</View>
`;

exports[`displays success toaster without text2 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#CD03EF",
      "borderRadius": 10.714285714285714,
      "marginBottom": 21.428571428571427,
      "opacity": 1,
      "padding": 21.428571428571427,
      "width": "80%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#ECEDEE",
        },
        {
          "fontSize": 34.285714285714285,
          "lineHeight": 51.42857142857143,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        [
          {
            "color": "#fff",
          },
          undefined,
        ],
      ]
    }
  >
    Click me
  </Text>
</View>
`;

exports[`displays toaster without text2 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#CD03EF",
      "borderRadius": 10.714285714285714,
      "marginBottom": 21.428571428571427,
      "opacity": 1,
      "padding": 21.428571428571427,
      "width": "80%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#ECEDEE",
        },
        {
          "fontSize": 34.285714285714285,
          "lineHeight": 51.42857142857143,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        [
          {
            "color": "#fff",
          },
          undefined,
        ],
      ]
    }
  >
    Click me
  </Text>
</View>
`;

exports[`displays warning toaster 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#CD03EF",
      "borderRadius": 10.714285714285714,
      "marginBottom": 21.428571428571427,
      "opacity": 1,
      "padding": 21.428571428571427,
      "width": "80%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#ECEDEE",
        },
        {
          "fontSize": 34.285714285714285,
          "lineHeight": 51.42857142857143,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        [
          {
            "color": "#fff",
          },
          undefined,
        ],
      ]
    }
  >
    Click me
  </Text>
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#CD03EF",
      "borderRadius": 10.714285714285714,
      "marginBottom": 21.428571428571427,
      "opacity": 1,
      "padding": 21.428571428571427,
      "width": "80%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#ECEDEE",
        },
        {
          "fontSize": 34.285714285714285,
          "lineHeight": 51.42857142857143,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        [
          {
            "color": "#fff",
          },
          undefined,
        ],
      ]
    }
  >
    Click me
  </Text>
</View>
`;
